### PR TITLE
FIP-113: End Departed Rari Founders vesting of TRIBE

### PR DESCRIPTION
**Pull request by thomas-waite**
# FIP-113: End Departed Rari Founders vesting of TRIBE

Implements the on-chain DAO vote part of https://tribe.fei.money/t/fip-113-end-departed-rari-founders-vesting-of-tribe/4355 

Specifically: 
1. Migrates the Rari infra team timelocks
2. Claws back the vesting TRIBE of Jai Bhavnani and David Lucid

## Timelock migration is performed by:
- Set the `beneficiary` of the existing timelocks to the Tribal Council timelock
- Deploy new Fei and Tribe timelocks for the Rari infra team
- Set the `beneficiary` of the Rari infra team timelocks to be the new Rari ops multisig
- Set the `clawbackAdmin` of the Rari infra team timelocks to be the Tribal Council timelock
- Set the start time as now, the time at which they are deployed
- Set the duration as the remaining duration of the existing timelocks (~580 days)
- Mint the amount of Fei and allocate Tribe to the timelocks as is currently left on the original timelocks

The Rari Infra timelocks are linear, so we can simply mint/allocate remaining funds there and set the duration to the remaining duration.

### PR Checklist - Feature (Proposal)

- [ ] All Tests Passing
- [ ] Proposal Added to ProposalsConfig
- [ ] Fork Block Correct
- [ ] Remove Any .only's on Tests
- [ ] Update Documentation If Needed
- [ ] Update Roles Config
- [ ] Proposal Submitted

### CODE REVIEW

File: proposals/dao/clawback.ts
Comment (thomas-waite) [inline proposals/dao/clawback.ts]: Will update with final numbers once Rari have claimed what is available on their timelocks

File: protocol-configuration/mainnetAddresses.ts
Comment (thomas-waite) [inline protocol-configuration/mainnetAddresses.ts]: To fill in when we deploy the new contracts

File: proposals/dao/clawback.ts
Comment (eswak) [inline proposals/dao/clawback.ts]: I think you need a 2nd factory to create `LinearTokenTimelock` for the FEI rari infra team vesting contract

File: proposals/dao/clawback.ts
Comment (eswak) [inline proposals/dao/clawback.ts]: the FEI timelock should be a `LinearTokenTimelock` there is no `delegate(address)` function on the FEI token contract

Comment (thomas-waite) [reply proposals/dao/clawback.ts]: Ah good point, oversight

File: proposals/description/clawback.ts
Comment (eswak) [inline proposals/description/clawback.ts]: todo: update with the new balance after rari infra team makes a last claim

File: proposals/description/clawback.ts
Comment (eswak) [inline proposals/description/clawback.ts]: todo: update with the new balance after rari infra team makes a last claim

**Comment by eswak** [comment]
just a minor change to do imo (change the factory for Rari infra team FEI's vesting), otherwise lgtm

edit: lol you commited the fix before I commented 💪

File: proposals/dao/clawback.ts
Comment (thomas-waite) [reply proposals/dao/clawback.ts]: done

File: proposals/description/clawback.ts
Comment (thomas-waite) [reply proposals/description/clawback.ts]: done

File: proposals/description/clawback.ts
Comment (thomas-waite) [reply proposals/description/clawback.ts]: done

File: protocol-configuration/mainnetAddresses.ts
Comment (ElliotFriedman) [inline protocol-configuration/mainnetAddresses.ts]: these should be deployed now so the DAO vote can be submitted

Comment (thomas-waite) [reply protocol-configuration/mainnetAddresses.ts]: Getting reviews and approval on this PR first before I deploy

File: proposals/dao/clawback.ts
Comment (thomas-waite) [reply proposals/dao/clawback.ts]: done

**Review by eswak — COMMENTED** [review]
lgtm (still need to deploy)

File: test/integration/tests/fip-38-tokemak.ts
Comment (thomas-waite) [inline test/integration/tests/fip-38-tokemak.ts:17]: Rolling over doesn't work for some reason when forking and having a proposal that requests a withdrawal. Erwan has the same issue. He's deleted on the release branch as tokemak will go away next week

Comment (thomas-waite) [reply test/integration/tests/fip-38-tokemak.ts:17]: Skipping for now, will come back to after this PR if important

File: test/integration/tests/tribeIncentivesEnd.ts
Comment (thomas-waite) [inline test/integration/tests/tribeIncentivesEnd.ts:62]: No longer relevant/work, proposal to end incentives has finished

File: proposals/description/clawback.ts
Comment (Joeysantoro) [inline proposals/description/clawback.ts]: the proposal should withdraw all the TRIBE from the timelock back to core after or at least approve some contract like the Tribal Council Timelock to do so

Comment (thomas-waite) [reply proposals/description/clawback.ts]: Have approved the TC timelock for 20M and added checks that it was approved and can transfer

**Review by Joeysantoro — APPROVED** [review]

File: proposals/dao/clawback.ts
Comment (ElliotFriedman) [inline proposals/dao/clawback.ts:59]: remove this commented code

File: proposals/dao/clawback.ts
Comment (ElliotFriedman) [inline proposals/dao/clawback.ts:62]: remove commented code

Comment (thomas-waite) [reply proposals/dao/clawback.ts:62]: Was leaving in in case I needed to do a redeploy (it's needed during deploy time) and will remove after

File: proposals/dao/clawback.ts
Comment (thomas-waite) [reply proposals/dao/clawback.ts:59]: Was leaving in in case I needed to do a redeploy (it's needed during deploy time) and will remove after

**Pull request merged**
